xarray/tests/test_dataset.py

```diff
@@ -2371,8 +2371,12 @@ def test_drop_index_labels(self):
             data.drop(DataArray(["a", "b", "c"]), dim="x", errors="ignore")
         assert_identical(expected, actual)
 
-        with raises_regex(ValueError, "does not have coordinate labels"):
-            data.drop_sel(y=1)
+        actual = data.drop_sel(y=[1])
+        expected = data.isel(y=[0, 2])
+        assert_identical(expected, actual)
```

Comment:
as mentioned above, I think `data.drop_sel(y=...)` should not return successfully, so the check for a `ValueError` was correct. The `KeyError` below is okay, though.